core/src/test/scala/org/apache/spark/scheduler/SchedulerIntegrationSuite.scala

```diff
@@ -391,17 +391,18 @@ private[spark] abstract class MockBackend(
    * scheduling.
    */
   override def reviveOffers(): Unit = {
-    val newTaskDescriptions = taskScheduler.resourceOffers(generateOffers()).flatten
-    // get the task now, since that requires a lock on TaskSchedulerImpl, to prevent individual
-    // tests from introducing a race if they need it
-    val newTasks = taskScheduler.synchronized {
-      newTaskDescriptions.map { taskDescription =>
-        val taskSet = taskScheduler.taskIdToTaskSetManager(taskDescription.taskId).taskSet
-        val task = taskSet.tasks(taskDescription.index)
-        (taskDescription, task)
-      }
-    }
-    synchronized {
+    // Need a lock on the entire scheduler to protect freeCores -- otherwise, multiple threads
+    // may make offers at the same time, though they are using the same set of freeCores.
+    taskScheduler.synchronized {
+      val newTaskDescriptions = taskScheduler.resourceOffers(generateOffers()).flatten
+      // get the task now, since that requires a lock on TaskSchedulerImpl, to prevent individual
+      // tests from introducing a race if they need it.
+      val newTasks =
```

Comment:
super nit but can you fix the spacing here, so the newTaskDescriptions.map is on the previous line after the "=", and then the closing bracket on line 405 will line up with the next line.